### PR TITLE
Update docsy version

### PR DESCRIPTION
**Pull request by geriom**
# Changes

[Changes Preview](https://docsy-update--tekton.netlify.app/docs/)

Docsy now has versioned releases and uses Hugo modules (a wraper around
Go modules). This PR update the website to be compatible with the latest
version of Docsy.

- Create go.mod and go.sum files for compatibility with Hugo modules.
- Delete checked-in theme and dependencies.
- Update base layout for compatibility with latest theme.
- Update packaje.json to match required dependencies and replace legacy
  values
- Cosmetic changes: Collapsed nav bar with animation to expand/collapse, change code highlighting theme to "manni" which matches the overall colors scheme better.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Review by afrittoli — COMMENTED** [review]
Thank you for this @geriom - the preview looks good to me - and if we find some small issues we can fix them on `main`.
This also unlocks further updates and API docs, so really looking forward to it.
/approve

**Review by AlanGreene — COMMENTED** [review]
I'm still working through the rest of the changes / reviewing the preview but so far it looks good. Couple of config items that were disabled in the preview branch that I think should be re-enabled now for production use.

**Comment by AlanGreene** [comment]
The only other thing I've found so far is that the links in the right sidebar are broken or incomplete. I'll continue reviewing this evening.

e.g. 

On the live site these work as expected
https://tekton.dev/docs/pipelines/container-contract/
image

They link to:
- https://github.com/tektoncd/pipeline/edit/main/docs/container-contract.md
- https://github.com/tektoncd/website/issues/new?title=Container%20Contract

On the new preview
https://deploy-preview-421--tekton.netlify.app/docs/pipelines/container-contract/
image
- the view / edit links point to the website repo instead of pipeline repo
- the create issue links don't include the page title

**Comment by AlanGreene** [comment]
Just noticed the header changes to blue when scrolling down the home page:
image

File: config.toml [hunk 1]
```diff
@@ -51,7 +49,7 @@ anchor = "smart"
 [services]
 [services.googleAnalytics]
 # Comment out the next line to disable GA tracking. Also disables the feature described in [params.ui.feedback].
-id = "UA-154682132-1"
+# id = "UA-154682132-1"
```
[the remark this answers is not in the document]
Comment (geriom) [reply config.toml]: Re-enabled

**Comment by geriom** [comment]
> The only other thing I've found so far is that the links in the right sidebar are broken or incomplete. I'll continue reviewing this evening.
> 

Fixed. I copied the old 'page-metadata-links' partial into Tekton to overwrite the default behavior and keep the current style.

> Just noticed the header changes to blue when scrolling down the home page.

Fixed. Added style that sets the background to 'dark' when scrolling.

@AlanGreene PTAL

**Comment by AlanGreene** [comment]
Looks great, thanks @geriom 
/lgtm

**Pull request merged**